blockchain/hash.go

```diff
@@ -26,3 +56,63 @@ type Marshaller interface {
 func HashSum(m Marshaller) Hash {
 	return sha256.Sum256(m.Marshal())
 }
+
+// CompareTo compares two hashes, it returns true if the operation of the first hash on the second hash specified by the comparator is true, and false otherwise
+func (h Hash) CompareTo(h2 Hash, comparator int) bool {
+	for i := HashLen - 1; i >= 0; i-- {
+		if h[i] > h2[i] {
+			return GreaterThan == comparator
+		} else if h[i] < h2[i] {
+			return LessThan == comparator
+		}
+	}
+	return EqualTo == comparator
+}
+
+// ParseString sets the hash to the the value represented by a hexadecimal string, and returns the hash
+func (h *Hash) ParseString(s string) *Hash {
+	if len(s) == 0 || len(s) == 1 || s[:2] != "0x" {
+		log.Error("Invalid hexadecimal string, must begin with 0x")
+		return nil
+	}
+
+	s = s[2:]
+	if len(s) > HashLen*2 {
+		log.Error("Hexadeximal string is too large")
+		return nil
+	}
+
+	if len(s)%2 != 0 {
+		s = "0" + s
+	}
+
+	if len(s)/2 < HashLen {
+		pre := ""
+		for i := 0; i < HashLen-len(s)/2; i++ {
+			pre += "00"
+		}
+		s = pre + s
+	}
+
+	var bytes, e = hex.DecodeString(s)
+
+	if e != nil {
+		log.Error(e)
+		return nil
+	}
+
+	for i := 0; i < HashLen; i++ {
+		h[i] = bytes[HashLen-1-i]
+	}
+
+	return h
+}
+
+// HexString returns the hexadecimal representation of the hash
+func (h Hash) HexString() string {
+	var hashBigEndian Hash
+	for i := 0; i < HashLen; i++ {
```

Comment:
I notice this pattern of inverting the hex string a couple of times, perhaps a helper function?